python/foglamp/common/service_record.py

```diff
@@ -25,11 +25,23 @@ class Type(IntEnum):
         Core = 2
         Southbound = 3
 
+    class Status(IntEnum):
+        """Enumeration for Service Status"""
+
+        Running = 1
+        Unregistered = 2
+        Failed = 3
+        Doubtful = 4
+
     class InvalidServiceType(Exception):
         # TODO: tell allowed service types?
         pass
 
-    __slots__ = ['_id', '_name', '_type', '_protocol', '_address', '_port', '_management_port', '_status']
+    class InvalidServiceStatus(Exception):
+        # TODO: tell allowed service status?
+        pass
+
+    __slots__ = ['_id', '_name', '_type', '_protocol', '_address', '_port', '_management_port', '_status', '_check_count']
```

Comment:
_check_count MUST not be an attribute of ServiceRecord entity.

Reply:
Nah. We need to keep the ok count with each service and there is no other proper logical location. Otherwise, we will have to maintain a separate reference list of check_counts, thereby making code error prone and adding unneccessary redundancy.

Reply:
Via commit https://github.com/foglamp/FogLAMP/pull/685/commits/f15be0d7674ccb6cea4e1a8550f44c7428cc4335, a separate dict is now being maintained to hold check counts and _check_count now removed from service_record.